File: MPI_with_C/hello_world.c
#include <stdio.h>
#include <mpi.h>

int main(int argc, char *argv[]){

    // vars to store process ID(rank) and number of total processes(size)
    int rank, size;

    // starts up MPI for the processes running
    MPI_Init(&argc, &argv);

    // populates rank and size for processes on MPI_COMM_WORLD (default)
    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
    MPI_Comm_size(MPI_COMM_WORLD, &size);
    
    // each process prints their ID
    printf("Hello World, I am rank %d of %d\n", rank, size);

    // finished MPI
    MPI_Finalize();
    
	return 0;
}

File: hello_world.py
from mpi4py import MPI

# populates rank and size for processes on MPI.COMM_WORLD (default)
comm = MPI.COMM_WORLD
rank = comm.Get_rank()
size = comm.Get_size()

print('Hello World I am rank: ' + str(rank) + ' of ' + str(size))
